src/modules/transaction/txUtils.ts

```diff
@@ -83,11 +83,21 @@ export async function getTransaction(
 
   const receipt = await eth.getTransactionReceipt(hash)
 
+  if (receipt == null) {
```

Comment:
I'm not sure about this, sometimes the tx can be dropped or reorg leaving it outside the main-chain. If I'm not wrong, the receipt will `null` too. Leaving a forever loading. 

Maybe if `receipt == null` we can also check for something else which we are doing at line 53.

What was the case that motivates the PR?

Reply:
If it would get dropped, then wouldn't it be checked later on and moved from "pending" to "dropped" status?

Motivation for this PR: I was sending transactions in my app and listening to later actions FETCH_TRANSACTION_SUCCESS and FETCH_TRANSACTION_FAILURE, to update/refresh my app status accordingly. It became a problem though when I was 
 wrongly having FETCH_TRANSACTION_FAILURE as if they were reverted, when they actually wasn't.

Reply:
Cool! do you have examples? 

`Dropped & replaced` yes, will happen what you said. But it was reorg and not included in the chain the tx will be a ghost one like it will never happen.

Reply:
I was just creating a transaction, and listening to updates by using selectors to grab pending/confirmed transactions from the redux state.

Not sure how to handle the reorg case. In this scenario, what would be the expected result? Following the original code logic + your logic, I think you imply that it should be marked as Reverted?  However maybe that is a different issue, in order to be sure if the tx was actually placed in the main-chain (ie. "confirmed ~100%"?) , there should be some check of the # of confirmations, and act on the tx result after asserting that enough confirmations have been achieved. 

The current change on this PR, just keeps the tx status as "Pending" (instead of wrongly "Reverted"), until more information is available to say otherwise.  Should this change also include logic about the block hash and # confirmations? This [SO answer](https://ethereum.stackexchange.com/a/53270) proposes a strategy for achieving this

Reply:
> Should this change also include logic about the block hash and # confirmations? This SO answer proposes a strategy for achieving this.

This is exactly what I was trying to explain.

Reply:
@tmilar is this still in your backlog?

Reply:
Hi @nachomazzara , I'm sorry but I'm not actively working on this project anymore. 
Though if it still matters to you let me know, I can make an effort to complete the PR when I have some free time.

Reply:
It would be great if you can complete it :)